lib/sdf-server/src/app.rs

```diff
@@ -0,0 +1,134 @@
+use std::sync::Arc;
+
+use axum::Router;
+use dal::{JwtPublicSigningKey, ServicesContext};
+use nats_multiplexer_client::MultiplexerClient;
+use si_posthog::PosthogClient;
+use telemetry::prelude::*;
+use tokio::sync::RwLock;
+use tokio_util::sync::CancellationToken;
+use tower_http::trace::TraceLayer;
+
+use crate::{
+    routes::routes, AppState, ApplicationRuntimeMode, WorkspacePermissions,
+    WorkspacePermissionsMode,
+};
+
+#[derive(Debug)]
+pub struct AxumApp(Router);
+
+impl AxumApp {
+    #[allow(clippy::too_many_arguments)]
+    #[inline]
+    pub fn from_services(
+        services_context: ServicesContext,
+        jwt_public_signing_key: JwtPublicSigningKey,
+        posthog_client: PosthogClient,
+        auth_api_url: impl AsRef<str>,
+        ws_multiplexer_client: MultiplexerClient,
+        crdt_multiplexer_client: MultiplexerClient,
+        create_workspace_permissions: WorkspacePermissionsMode,
+        create_workspace_allowlist: Vec<WorkspacePermissions>,
+        application_runtime_mode: Arc<RwLock<ApplicationRuntimeMode>>,
+        shutdown_token: CancellationToken,
+    ) -> Self {
+        Self::inner_from_services(
+            services_context,
+            jwt_public_signing_key,
+            posthog_client,
+            auth_api_url,
+            false,
+            ws_multiplexer_client,
+            crdt_multiplexer_client,
+            create_workspace_permissions,
+            create_workspace_allowlist,
+            application_runtime_mode,
+            shutdown_token,
+        )
+    }
+
+    // TODO(fnichol): This really smells, we should not be flipping the app behavior differently if
+    // we're in a testing scenario. What is the configurable differences necessary to drive the app
+    // under test? *That* is how the app should be configured--no `for_tests` bool.
+    //
+    // Until that is refactored though...this constructor remains.
+    #[allow(clippy::too_many_arguments)]
+    #[inline]
+    pub fn from_services_for_tests(
```

Comment:
Frankly, I think it's just a wrapper and I think we can dismantle it later. Of course, there may be a config change or two between prod and testing, but I think this is mainly old patterns.